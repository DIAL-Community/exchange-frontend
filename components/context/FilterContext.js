import { createContext, useEffect, useState } from 'react'
import { useRouter } from 'next/router'

const FilterContext = createContext()
const FilterDispatchContext = createContext()

const FilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')
  // Task tracker context only
  const [showFailedOnly, setShowFailedOnly] = useState(false)
  const [showGovStackOnly, setShowGovStackOnly] = useState(false)
  const [showMature, setShowMature] = useState(false)

  const [categoryTypes, setCategoryTypes] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])

  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [tags, setTags] = useState([])
  const [datasetTypes, setDatasetTypes] = useState([])

  const [aggregators, setAggregators] = useState([])
  const [operators, setOperators] = useState([])
  const [services, setServices] = useState([])

  const [years, setYears] = useState([])

  const [products, setProducts] = useState([])

  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [showClosed, setShowClosed] = useState(false)

  const [aggregator, setAggregator] = useState(false)
  const [endorser, setEndorser] = useState(false)
  const [endorserLevel, setEndorserLevel] = useState('')

  const [specialties, setSpecialties] = useState([])
  const [certifications, setCertifications] = useState([])

  const [isEndorsed, setIsEndorsed] = useState(false)
  const [endorsers, setEndorsers] = useState([])
  const [licenseTypes, setLicenseTypes] = useState([])
  const [isLinkedWithDpi, setIsLinkedWithDpi] = useState(false)
  const [showDpgaOnly, setShowDpgaOnly] = useState(false)

  const [softwareCategories, setSoftwareCategories] = useState([])
  const [softwareFeatures, setSoftwareFeatures] = useState([])
  const [comparedProducts, setComparedProducts] = useState([])
  const [productStage, setProductStage] = useState(null)

  const [showBeta, setShowBeta] = useState(false)

  const router = useRouter()
  useEffect(() => {
    const handleRouteChange = (url) => {
      // Transitioning to other pages (pathname changing).
      if (url.indexOf(router.pathname) < 0) {
        setSearch('')
        setShowFailedOnly(false)
        setShowGovStackOnly(false)
        setShowMature(false)

        setCategoryTypes([])
        setSdgs([])
        setUseCases([])
        setWorkflows([])

        setOrigins([])
        setCountries([])
        setSectors([])
        setOrganizations([])
        setTags([])
        setDatasetTypes([])

        setAggregators([])
        setOperators([])
        setServices([])

        setYears([])

        setProducts([])

        setBuildingBlocks([])
        setShowClosed(false)

        setAggregator(false)
        setEndorser(false)
        setEndorserLevel('')

        setSpecialties([])
        setCertifications([])

        setIsEndorsed(false)
        setEndorsers([])
        setLicenseTypes([])
        setIsLinkedWithDpi(false)
        setShowDpgaOnly(false)

        setSoftwareCategories([])
        setSoftwareFeatures([])
        setComparedProducts([])
        setProductStage('')

        setShowBeta(false)
      }
    }

    router.events.on('routeChangeComplete', handleRouteChange)

    return () => {
      router.events.off('routeChangeComplete', handleRouteChange)
    }
  }, [router.events, router.asPath, router.pathname])

  const valueProps = {
    search,

    showFailedOnly,
    showGovStackOnly,
    showMature,
    categoryTypes,
    sdgs,
    useCases,
    workflows,

    countries,
    datasetTypes,
    organizations,
    origins,
    sectors,
    tags,

    aggregators,
    operators,
    services,

    years,

    products,

    buildingBlocks,
    showClosed,

    aggregator,
    endorser,
    endorserLevel,
    specialties,
    certifications,

    isEndorsed,
    endorsers,
    licenseTypes,
    isLinkedWithDpi,
    showDpgaOnly,
    softwareCategories,
    softwareFeatures,
    comparedProducts,
    productStage,

    showBeta
  }

  const dispatchProps = {
    setSearch,
    setShowFailedOnly,
    setShowGovStackOnly,
    setShowMature,
    setCategoryTypes,
    setSdgs,
    setUseCases,
    setWorkflows,

    setCountries,
    setDatasetTypes,
    setOrganizations,
    setOrigins,
    setSectors,
    setTags,

    setAggregators,
    setOperators,
    setServices,

    setYears,

    setProducts,

    setBuildingBlocks,
    setShowClosed,

    setAggregator,
    setEndorser,
    setEndorserLevel,
    setSpecialties,
    setCertifications,

    setIsEndorsed,
    setEndorsers,
    setLicenseTypes,
    setIsLinkedWithDpi,
    setShowDpgaOnly,
    setSoftwareCategories,
    setSoftwareFeatures,
    setComparedProducts,
    setProductStage,

    setShowBeta
  }

  return (
    <FilterContext.Provider value={{ ...valueProps }}>
      <FilterDispatchContext.Provider value={{ ...dispatchProps }}>
        {children}
      </FilterDispatchContext.Provider>
    </FilterContext.Provider>
  )
}

export { FilterProvider, FilterContext, FilterDispatchContext }
