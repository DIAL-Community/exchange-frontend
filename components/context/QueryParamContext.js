import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
