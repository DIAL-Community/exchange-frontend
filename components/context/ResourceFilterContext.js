import React, { createContext, useState } from 'react'

const ResourceFilterContext = createContext()
const ResourceFilterDispatchContext = createContext()

const ResourceFilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')
  const [resourceTypes, setResourceTypes] = useState([])
  const [resourceTopics, setResourceTopics] = useState([])
  const [resourceCountries, setResourceCountries] = useState([])

  const resourceFilterValues = {
    search,
    resourceTypes,
    resourceTopics,
    resourceCountries
  }
  const resourceFilterDispatchValues = {
    setSearch,
    setResourceTypes,
    setResourceTopics,
    setResourceCountries
  }

  return (
    <ResourceFilterContext.Provider value={resourceFilterValues}>
      <ResourceFilterDispatchContext.Provider value={resourceFilterDispatchValues}>
        {children}
      </ResourceFilterDispatchContext.Provider>
    </ResourceFilterContext.Provider>
  )
}

export {
  ResourceFilterProvider,
  ResourceFilterContext,
  ResourceFilterDispatchContext
}
