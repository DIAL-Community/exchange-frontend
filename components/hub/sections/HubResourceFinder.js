import { FormattedMessage } from 'react-intl'
import HubResources from './HubResources'

const HubResourceFinder = () => {
  return (
    <>
      <img className='h-80 w-full object-cover' alt='DIAL DPI Resource Hub' src='/images/hero-image/hub-hero.svg'/>
      <div className='absolute w-full left-1/2 -translate-x-1/2 min-h-[20rem]' style={{ top: 'var(--ui-header-height)' }}>
        <div className='text-2xl text-center text-dial-cotton pt-20 pb-10 uppercase mx-auto max-w-prose'>
          <FormattedMessage
            id='dpi.resourceFinder.title'
            values={{
              break: () => <br />
            }}
          />
        </div>
        <div className='text-lg text-center text-dial-cotton mx-auto max-w-prose'>
          <FormattedMessage
            id='dpi.resourceFinder.subtitle'
            values={{
              break: () => <br />
            }}
          />
        </div>
      </div>
      <HubResources />
    </>
  )
}

export default HubResourceFinder
