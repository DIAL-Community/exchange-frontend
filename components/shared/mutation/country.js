import { gql } from '@apollo/client'

export const DELETE_COUNTRY = gql`
  mutation DeleteCountry($id: ID!) {
    deleteCountry(id: $id) {
      country {
        id
        slug
        name
      }
      errors
    }
  }
`

export const CREATE_COUNTRY = gql`
  mutation CreateCountry(
    $name: String!
    $slug: String!
    $description: String
  ) {
    createCountry(
      name: $name
      slug: $slug
      description: $description
    ) {
        country {
          id
          name
          slug
          code
          codeLonger
          description
        }
        errors
      }
    }
`
