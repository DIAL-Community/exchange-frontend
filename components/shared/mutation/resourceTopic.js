import { gql } from '@apollo/client'

export const DELETE_RESOURCE_TOPIC = gql`
  mutation DeleteResourceTopic($id: ID!) {
    deleteResourceTopic(id: $id) {
      resourceTopic {
        id
      }
      errors
    }
  }
`

export const CREATE_RESOURCE_TOPIC = gql`
  mutation CreateResourceTopic(
    $name: String!
    $slug: String!
    $imageFile: Upload
    $description: String
    $parentTopicId: ID
  ) {
    createResourceTopic(
      name: $name
      slug: $slug
      imageFile: $imageFile
      description: $description
      parentTopicId: $parentTopicId
    ) {
      resourceTopic {
        id
        name
        slug
        imageFile
        parentTopic {
          id
          name
          slug
        }
        resourceTopicDescription {
          id
          description
          locale
        }
      }
      errors
    }
  }
`
