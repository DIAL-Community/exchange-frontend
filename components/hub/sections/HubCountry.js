import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { useQuery } from '@apollo/client'
import { Error, Loading, NotFound } from '../../shared/FetchStatus'
import { DPI_COUNTRY_DETAIL_QUERY } from '../../shared/query/country'
import HubCountryDetail from '../fragments/HubCountryDetail'
import HubBreadcrumb from './HubBreadcrumb'

const HubCountry = ({ slug }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { loading, error, data } = useQuery(DPI_COUNTRY_DETAIL_QUERY, {
    variables: { slug }
  })

  if (loading) {
    return <Loading />
  } else if (error) {
    return <Error />
  } else if (!data?.country) {
    return <NotFound />
  }

  const { country } = data

  const slugNameMapping = (() => {
    const map = {}
    map[country.slug] = country.name

    return map
  })()

  return (
    <div className='flex flex-col'>
      <img className='h-80 w-full object-cover' alt='DIAL DPI Resource Hub' src='/images/hero-image/hub-hero.svg'/>
      <div className='absolute w-full left-1/2 -translate-x-1/2 min-h-[20rem]' style={{ top: 'var(--ui-header-height)' }}>
        <div className='max-w-catalog mx-auto'>
          <div className='px-4 lg:px-8 xl:px-56 text-dial-gray'>
            <HubBreadcrumb slugNameMapping={slugNameMapping} />
          </div>
        </div>
        <div className='text-2xl text-center text-dial-cotton pt-20 pb-4 mx-auto max-w-prose'>
          {country.name}
        </div>
        <img
          src={`https://flagcdn.com/${country.code.toLowerCase()}.svg`}
          alt={format('ui.country.logoAlt', { countryName: country.code })}
          className='h-16 mx-auto'
        />
      </div>
      <HubCountryDetail country={country} />
    </div>
  )
}

export default HubCountry
