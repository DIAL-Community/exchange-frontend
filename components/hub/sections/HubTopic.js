import parse from 'html-react-parser'
import { useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../lib/apolloClient'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../../shared/GraphQueryHandler'
import { RESOURCE_TOPIC_DETAIL_QUERY } from '../../shared/query/resourceTopic'
import HubTopicDetail from '../fragments/HubTopicDetail'
import HubBreadcrumb from './HubBreadcrumb'

const HubTopic = ({ slug, pageNumber, onClickHandler }) => {
  const { loading, error, data } = useQuery(RESOURCE_TOPIC_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.resourceTopic) {
    return handleMissingData()
  }

  const { resourceTopic } = data

  const slugNameMapping = (() => {
    const map = {}
    map[resourceTopic.slug] = resourceTopic.name

    return map
  })()

  return (
    <div className='flex flex-col gap-6 pb-12 max-w-catalog mx-auto'>
      <img
        className='h-32 w-full object-cover'
        alt='DIAL Resource Hub - Resource Topic'
        src='/images/hero-image/hub-topic.svg'
      />
      <div className='absolute w-full left-1/2 -translate-x-1/2' style={{ top: 'var(--ui-header-height)' }}>
        <div className='max-w-catalog mx-auto py-2'>
          <div className='px-4 lg:px-8 xl:px-56 text-dial-gray'>
            <HubBreadcrumb slugNameMapping={slugNameMapping} />
          </div>
          <div className='px-4 lg:px-8 xl:px-56 py-8'>
            <div className='text-2xl text-dial-cotton uppercase max-w-prose'>
              {resourceTopic.name}
            </div>
          </div>
        </div>
      </div>
      <div className='px-4 lg:px-8 xl:px-56 min-h-[60vh] lg:min-h-[65vh]'>
        <div className='flex flex-col gap-y-6'>
          <div className='text-dial-stratos max-w-5xl text-justify'>
            {resourceTopic?.resourceTopicDescription && parse(resourceTopic.resourceTopicDescription?.description)}
          </div>
          <HubTopicDetail resourceTopic={resourceTopic} pageNumber={pageNumber} onClickHandler={onClickHandler} />
        </div>
      </div>
    </div>
  )
}

export default HubTopic
