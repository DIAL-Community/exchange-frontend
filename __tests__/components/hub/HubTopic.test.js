import { screen } from '@testing-library/dom'
import { FilterProvider } from '../../../components/context/FilterContext'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { ResourceFilterProvider } from '../../../components/context/ResourceFilterContext'
import HubTopic from '../../../components/hub/sections/HubTopic'
import { COUNTRIES_WITH_RESOURCES_SEARCH_QUERY } from '../../../components/shared/query/country'
import {
  PAGINATED_RESOURCES_QUERY, RESOURCE_PAGINATION_ATTRIBUTES_QUERY, RESOURCE_TYPE_SEARCH_QUERY
} from '../../../components/shared/query/resource'
import { RESOURCE_TOPIC_DETAIL_QUERY } from '../../../components/shared/query/resourceTopic'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import { resourcePagination, resources } from './data/HubCountry.data'
import { resourceTypes } from './data/HubResourceFinder.data'
import { countriesWithResources, resourceTopic } from './data/HubTopic.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the opportunity detail page.', () => {
  const mockResourceTopic = generateMockApolloData(
    RESOURCE_TOPIC_DETAIL_QUERY,
    {
      'slug': 'data-privacy'
    },
    null,
    resourceTopic
  )

  const mockCountriesWithResources = generateMockApolloData(
    COUNTRIES_WITH_RESOURCES_SEARCH_QUERY,
    {
      'search': ''
    },
    null,
    countriesWithResources
  )

  const mockResourceTypes = generateMockApolloData(
    RESOURCE_TYPE_SEARCH_QUERY,
    {
      'search': ''
    },
    null,
    resourceTypes
  )

  const mockResources = generateMockApolloData(
    PAGINATED_RESOURCES_QUERY,
    {
      'search': '',
      'countries': [],
      'resourceTypes': [],
      'resourceTopics': ['Data Privacy'],
      'limit': 6,
      'offset': 0
    },
    null,
    resources
  )

  const mockResourcePagination = generateMockApolloData(
    RESOURCE_PAGINATION_ATTRIBUTES_QUERY,
    {
      'search': '',
      'countries': [],
      'resourceTypes': [],
      'resourceTopics': ['Data Privacy']
    },
    null,
    resourcePagination
  )

  test('Should render topic tiles page.', async () => {
    const { container } = render(
      <CustomMockedProvider
        mocks={[
          mockResourceTypes,
          mockResourceTopic,
          mockCountriesWithResources,
          mockResources,
          mockResourcePagination
        ]}
      >
        <QueryParamContextProvider>
          <ResourceFilterProvider>
            <FilterProvider>
              <HubTopic slug='data-privacy' pageNumber={0} onClickHandler={() => { }} />
            </FilterProvider>
          </ResourceFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('Data Privacy')).toBeInTheDocument()
    expect(await screen.findByText('Stats SL Open Data Dashboard')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
