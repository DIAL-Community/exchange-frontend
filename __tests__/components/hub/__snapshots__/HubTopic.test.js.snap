// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the opportunity detail page. Should render topic tiles page. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="flex flex-col gap-6 pb-12 max-w-catalog mx-auto"
  >
    <img
      alt="DIAL Resource Hub - Resource Topic"
      class="h-32 w-full object-cover"
      src="/images/hero-image/hub-topic.svg"
    />
    <div
      class="absolute w-full left-1/2 -translate-x-1/2"
    >
      <div
        class="max-w-catalog mx-auto py-2"
      >
        <div
          class="px-4 lg:px-8 xl:px-24 3xl:px-56 text-dial-gray"
        >
          <div
            class="whitespace-nowrap text-ellipsis overflow-hidden text-sm"
          >
            <a
              class="inline"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
        <div
          class="px-4 lg:px-8 xl:px-24 3xl:px-56 py-8"
        >
          <div
            class="text-2xl text-dial-cotton uppercase max-w-prose"
          >
            Data Privacy
          </div>
        </div>
      </div>
    </div>
    <div
      class="px-4 lg:px-8 xl:px-24 3xl:px-56 min-h-[60vh] lg:min-h-[65vh]"
    >
      <div
        class="flex flex-col gap-y-6"
      >
        <div
          class="text-dial-stratos max-w-5xl text-justify"
        />
        <div
          class="flex flex-col gap-5"
        >
          <div
            class="text-lg text-center"
          >
            Reports and Articles
          </div>
          <div
            class="flex flex-col gap-y-4 mb-6"
          >
            <div
              class="flex flex-col gap-y-2"
            >
              <div
                class="text-sm font-semibold text-dial-sapphire"
              >
                Filter your results by
                :
              </div>
            </div>
            <div
              class="grid grid-cols-1 xl:grid-cols-3 gap-8"
            >
              <div
                class="rounded text-sm text-dial-gray-dark my-auto css-1uaxf62-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-:r0:-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                  role="log"
                />
                <div
                  class="react-select__control css-afrqkl-control"
                >
                  <div
                    class="react-select__value-container css-17t7gkf-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-5rjclb-placeholder"
                      id="react-select-:r0:-placeholder"
                    >
                      Filter by Resource Type
                    </div>
                    <div
                      class="react-select__input-container css-qbdosj-Input"
                      data-value=""
                    >
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-describedby="react-select-:r0:-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Filter by Resource Type"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="react-select__input"
                        id="async-select-undefined"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__loading-indicator css-pogzpp-loadingIndicator"
                    >
                      <span
                        class="css-1xtdfmb-LoadingDot"
                      />
                      <span
                        class="css-zoievk-LoadingDot"
                      />
                      <span
                        class="css-x748d8-LoadingDot"
                      />
                    </div>
                    <span
                      class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex flex-col gap-y-3"
              >
                <div
                  class="rounded text-sm text-dial-gray-dark my-auto css-1uaxf62-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-:r1:-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="react-select__control css-afrqkl-control"
                  >
                    <div
                      class="react-select__value-container css-17t7gkf-ValueContainer"
                    >
                      <div
                        class="react-select__placeholder css-5rjclb-placeholder"
                        id="react-select-:r1:-placeholder"
                      >
                        Filter by Country
                      </div>
                      <div
                        class="react-select__input-container css-qbdosj-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-describedby="react-select-:r1:-placeholder"
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-label="Filter by Country"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="react-select__input"
                          id="async-select-undefined"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__loading-indicator css-pogzpp-loadingIndicator"
                      >
                        <span
                          class="css-1xtdfmb-LoadingDot"
                        />
                        <span
                          class="css-zoievk-LoadingDot"
                        />
                        <span
                          class="css-x748d8-LoadingDot"
                        />
                      </div>
                      <span
                        class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="flex flex-row items-center justify-center gap-1 relative"
              >
                <div
                  class="ml-auto w-full max-w-prose"
                >
                  <label
                    class="relative"
                    for="search-bar"
                  >
                    <input
                      class="w-full border text-sm"
                      id="search-bar"
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                    <svg
                      class="absolute top-1/2 transform -translate-y-1/2 right-2"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-8"
          >
            <div
              class="relative"
            >
              <div
                class="rounded-lg min-h-[6rem] false"
              >
                <div
                  class="min-w-80 pb-4 mx-auto flex flex-col gap-3"
                >
                  <div
                    class="w-full flex justify-center items-center"
                  >
                    <a
                      class="w-full h-[267px]"
                      href="/hub/resources/stats-sl-open-data-dashboard"
                    >
                      <img
                        alt="Logo of Resource."
                        class="aspect-auto h-[267px]"
                        src="http://localhost:3000//assets/resources/resource-placeholder.svg"
                      />
                    </a>
                  </div>
                  <a
                    href="/hub/resourcesstats-sl-open-data-dashboard"
                  >
                    <div
                      class="text-lg font-semibold text-dial-stratos"
                    >
                      Stats SL Open Data Dashboard
                    </div>
                  </a>
                  <a
                    href="/hub/resourcesstats-sl-open-data-dashboard"
                  >
                    <div
                      class="line-clamp-4 text-dial-stratos text-sm"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="relative"
            >
              <div
                class="rounded-lg min-h-[6rem] false"
              >
                <div
                  class="min-w-80 pb-4 mx-auto flex flex-col gap-3"
                >
                  <div
                    class="w-full flex justify-center items-center"
                  >
                    <a
                      class="w-full h-[267px]"
                      href="/hub/resources/sierra-leone-data-portal"
                    >
                      <img
                        alt="Logo of Resource."
                        class="aspect-auto h-[267px]"
                        src="http://localhost:3000//assets/resources/resource-placeholder.svg"
                      />
                    </a>
                  </div>
                  <a
                    href="/hub/resourcessierra-leone-data-portal"
                  >
                    <div
                      class="text-lg font-semibold text-dial-stratos"
                    >
                      Sierra Leone Data Portal
                    </div>
                  </a>
                  <a
                    href="/hub/resourcessierra-leone-data-portal"
                  >
                    <div
                      class="line-clamp-4 text-dial-stratos text-sm"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
          Fetching data ...
        </div>
      </div>
    </div>
  </div>
</div>
`;
